relayer-cosmos/src/cosmos/core/types/telemetry.rs

```diff
@@ -0,0 +1,126 @@
+use alloc::sync::Arc;
+use ibc_relayer_framework::core::traits::runtime::telemetry::*;
+use opentelemetry::{
+    metrics::{Counter, Meter, UpDownCounter, ValueRecorder},
+    KeyValue,
+};
+use std::collections::HashMap;
+use std::sync::Mutex;
+
+pub struct TelemetryState {
+    pub meter: Meter,
+    pub counters: HashMap<String, Counter<u64>>,
+    pub value_recorders: HashMap<String, ValueRecorder<u64>>,
+    pub updown_counters: HashMap<String, UpDownCounter<i64>>,
+}
+
+#[derive(Clone)]
+pub struct CosmosTelemetry {
+    pub telemetry_state: Arc<Mutex<TelemetryState>>,
+}
+
+impl CosmosTelemetry {
+    pub fn new(telemetry_state: Arc<Mutex<TelemetryState>>) -> Self {
+        Self { telemetry_state }
+    }
+}
+
+impl HasLabel for CosmosTelemetry {
+    type Label = KeyValue;
+    fn new_label(key: &str, value: &str) -> Self::Label {
+        KeyValue::new(key.to_string(), value.to_string())
+    }
+}
+
+impl HasMetric<TelemetryCounter> for CosmosTelemetry {
```

Comment:
We can define a `OfaTelemetryCounter` that implements the preset metrics in `BasicTelemetryContext`. Something like:

```
pub trait OfaTelemetry {
  fn update_counter_metric(&self, ...);
  fn update_value_recorder_metric(&self, ...);
  fn update_up_down_counter_metric(&self, ...);
}
```

That way, `CosmosTelemetry` can implement only `OfaTelemetry` instead of implementing multiple `HasMetric` traits.